src/DocumentationBrowserViewExtension/DocumentationBrowserViewModel.cs

```diff
@@ -186,6 +199,59 @@ private void HandleLocalResource(OpenDocumentationLinkEventArgs e)
             this.shouldLoadDefaultContent = false;
         }
 
+        private void WatchMdFile(string mdLink)
+        {
+            if (string.IsNullOrWhiteSpace(mdLink))
+                return;
+
+            var fileName = Path.GetFileNameWithoutExtension(mdLink);
+            if (!packageManagerDoc.ContainsAnnotationDoc(fileName))
+                return;
+
+            markdownFileWatcher = new FileSystemWatcher(Path.GetDirectoryName(mdLink))
+            {
+                Filter = Path.GetFileName(mdLink),
+                EnableRaisingEvents = true
+            };
+            markdownFileWatcher.Changed += OnCurrentMdFileChanged;
+        }
+
+        private void UnsubscribeMdWatcher()
+        {
+            if (markdownFileWatcher is null)
+                return;
+
+            markdownFileWatcher.Changed -= OnCurrentMdFileChanged;
+        }
+
+        // Its a known issue that FileWatchers raises the same event twice when a file is modified from some programs.
+        // https://stackoverflow.com/questions/1764809/filesystemwatcher-changed-event-is-raised-twice
+        private void OnCurrentMdFileChanged(object sender, FileSystemEventArgs e)
+        {
+            if (e.FullPath != link.OriginalString)
+                return;
+            if (!(openDocumentationLinkEventArgs is OpenNodeAnnotationEventArgs))
+                return;
+
+            var nodeAnnotationArgs = openDocumentationLinkEventArgs as OpenNodeAnnotationEventArgs;
+            this.content = CreateNodeAnnotationContent(nodeAnnotationArgs);
+            this.Link = new Uri(e.FullPath);
+        }
+
+        private string CreateNodeAnnotationContent(OpenNodeAnnotationEventArgs e)
+        {
+            var writer = new StringWriter();
```

Comment:
@sm6srw @SHKnudsen this writer implements IDisposable, it should probably be wrapped in a using, can it be fixed?

Reply:
Sorry, missed this. I will take a look.